In [1]:
import pandas as pd
import numpy as np


# Section 1: Exploratory Data Analysis

### All questions in this section pertain to two of the datasets listed above in Exam Materials 

### (person_level_data.csv and household_level_data.csv), with the data_dictionary.txt file providing the metadata needed to understand the contents and formats of each .csv. 


### Please note that the data is still in its raw, original form and will require some cleaning and/or standardization.


Please write your answers in this document, and include your code as part1_code.txt.
Questions


Please calculate the following summary statistics using the data sets listed above:


1.	Calculate the average weekly hours worked among men and women.
2.	List the three households (i.e. household_ids) with the highest per person incomes. 
3.	List the three individuals (i.e. person_ids) with the highest personal income, and the dollar amounts of their personal incomes.
4.	Calculate the 80th percentile of age among people in each of the five marital categories.


In [2]:
person = pd.read_csv('person_level_data.csv')
household = pd.read_csv('household_level_data.csv')

person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status     sex     when_last_worked worked_last_week  \
0        married   not_student  FEMALE         in_last_year           worked   
1        married   not_student       M         in_last_year           worked   
2        married   not_student    MALE  over_five_years_ago     did_not_work   
3        married   not_student       F         in_last_year     did_not_work   
4        married   not_student  FEMALE  over_five_years_ago     not_reported   

   ... race_native_american  race_asian  race_black  race_native_hawaiian  \
0  ...                    0           0           0                     0   
1  ...                    0           0           0                     0   
2  ...                    0           0           0                     0   
3  ...                    0           0           0                     0   
4  ...                    0           0           0                     0   

   race_pacific_islander  race_white  race_other  uninsured  household_id  \
0                      0           1           0          0             5   
1                      0           1           0          0             5   
2                      0           1           0          0             9   
3                      0           1           0          0             9   
4                      0           1           0          0            22   

   hh_income_pct  
0       0.389078  
1       0.610922  
2       0.735381  
3       0.264619  
4       0.250230  

[5 rows x 21 columns]

In [4]:
person.sex.value_counts()

FEMALE    7223
MALE      6735
F         2056
M         1930
female    1017
male       936
Name: sex, dtype: int64

In [5]:
# Combining the different values for male and female
person.sex = person.sex.replace({'FEMALE': 'F'})
person.sex = person.sex.replace({'female': 'F'})
person.sex = person.sex.replace({'MALE': 'M'})
person.sex = person.sex.replace({'male': 'M'})
person.sex.value_counts()


F    10296
M     9601
Name: sex, dtype: int64

In [6]:
person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status sex     when_last_worked worked_last_week  ...  \
0        married   not_student   F         in_last_year           worked  ...   
1        married   not_student   M         in_last_year           worked  ...   
2        married   not_student   M  over_five_years_ago     did_not_work  ...   
3        married   not_student   F         in_last_year     did_not_work  ...   
4        married   not_student   F  over_five_years_ago     not_reported  ...   

  race_native_american  race_asian  race_black  race_native_hawaiian  \
0                    0           0           0                     0   
1                    0           0           0                     0   
2                    0           0           0                     0   
3                    0           0           0                     0   
4                    0           0           0                     0   

   race_pacific_islander  race_white  race_other  uninsured  household_id  \
0                      0           1           0          0             5   
1                      0           1           0          0             5   
2                      0           1           0          0             9   
3                      0           1           0          0             9   
4                      0           1           0          0            22   

   hh_income_pct  
0       0.389078  
1       0.610922  
2       0.735381  
3       0.264619  
4       0.250230  

[5 rows x 21 columns]

In [21]:
person.corr()

person_id       age  weekly_hours_worked  \
person_id               1.000000 -0.026801             0.013030   
age                    -0.026801  1.000000            -0.375180   
weekly_hours_worked     0.013030 -0.375180             1.000000   
race_native_american    0.005133 -0.010851            -0.007334   
race_asian             -0.008450 -0.066629             0.019717   
race_black             -0.003095 -0.040272            -0.094431   
race_native_hawaiian   -0.000857 -0.000546            -0.008694   
race_pacific_islander   0.016734 -0.017166             0.008030   
race_white              0.010829  0.097798             0.053821   
race_other             -0.015141 -0.091292             0.017906   
uninsured               0.006827 -0.095769            -0.034472   
household_id            0.999994 -0.026722             0.013010   
hh_income_pct           0.010970  0.216666             0.181575   

                       race_native_american  race_asian  race_black  \
person_id                          0.005133   -0.008450   -0.003095   
age                               -0.010851   -0.066629   -0.040272   
weekly_hours_worked               -0.007334    0.019717   -0.094431   
race_native_american               1.000000   -0.011480    0.002821   
race_asian                        -0.011480    1.000000   -0.068727   
race_black                         0.002821   -0.068727    1.000000   
race_native_hawaiian              -0.001187    0.010557    0.004874   
race_pacific_islander             -0.002190    0.070574    0.001152   
race_white                        -0.046694   -0.442792   -0.688728   
race_other                        -0.000921   -0.033883   -0.061832   
uninsured                          0.010277    0.012967    0.050898   
household_id                       0.005135   -0.008472   -0.003112   
hh_income_pct                     -0.006384   -0.042848    0.061661   

                       race_native_hawaiian  race_pacific_islander  \
person_id                         -0.000857               0.016734   
age                               -0.000546              -0.017166   
weekly_hours_worked               -0.008694               0.008030   
race_native_american              -0.001187              -0.002190   
race_asian                         0.010557               0.070574   
race_black                         0.004874               0.001152   
race_native_hawaiian               1.000000              -0.000464   
race_pacific_islander             -0.000464               1.000000   
race_white                        -0.032888              -0.047468   
race_other                        -0.003168               0.012060   
uninsured                          0.007793              -0.001545   
household_id                      -0.000915               0.016728   
hh_income_pct                      0.005432              -0.015299   

                       race_white  race_other  uninsured  household_id  \
person_id                0.010829   -0.015141   0.006827      0.999994   
age                      0.097798   -0.091292  -0.095769     -0.026722   
weekly_hours_worked      0.053821    0.017906  -0.034472      0.013010   
race_native_american    -0.046694   -0.000921   0.010277      0.005135   
race_asian              -0.442792   -0.033883   0.012967     -0.008472   
race_black              -0.688728   -0.061832   0.050898     -0.003112   
race_native_hawaiian    -0.032888   -0.003168   0.007793     -0.000915   
race_pacific_islander   -0.047468    0.012060  -0.001545      0.016728   
race_white               1.000000   -0.395127  -0.115375      0.010861   
race_other              -0.395127    1.000000   0.138760     -0.015167   
uninsured               -0.115375    0.138760   1.000000      0.006837   
household_id             0.010861   -0.015167   0.006837      1.000000   
hh_income_pct            0.008060   -0.070705  -0.052102      0.011040   

                       hh_income_pct  
person_id                   0.01

## Calculate the average weekly hours worked among men and women.

In [5]:
person.groupby('sex').weekly_hours_worked.mean()

sex
F         21.485409
FEMALE    21.815174
M         28.749741
MALE      28.453304
female    21.569322
male      30.064103
Name: weekly_hours_worked, dtype: float64

### On average men worked 28.67 hours

### On average women worked 21.73 hours

## List the three households (i.e. household_ids) with the highest per person incomes.

In [17]:
household.head()


household_id      variable  value
0            5     hh_income  90334
1            5       hh_size      2
2            5  num_children      0
3            9     hh_income  66216
4            9       hh_size      2

In [6]:
household = household.pivot(index='variable', columns='household_id', values='value')




In [20]:
household ['per_person'] = household.hh_income/household.hh_size

household.sort_values(by='per_person', ascending=False)



variable        hh_income hh_size num_children per_person
household_id                                             
38870              840245       1            0     840245
24934         1.11541e+06       2            0     557704
10110         1.10937e+06       2            0     554686
40137         1.04045e+06       2            0     520224
53260              518385       1            0     518385
...                   ...     ...          ...        ...
49062                   1       1            1          1
35019                   1       1            1          1
34684                   1       1            1          1
34242                   1       1            1          1
36540                   1       1            1          1

[10986 rows x 4 columns]

###  The three households with the highest per person incomes found by dividing the total income by family size are,

### household id, 38870
### household id, 24934
### household id, 10110

## List the three individuals (i.e. person_ids) with the highest personal income, and the dollar amounts of their personal incomes.

Personal id = 23355, Income = $1060273


Personal id = 89429, Income = $840245


Personal id = 82159, Income = $815297

In [8]:
pd.set_option('display.max_columns', 500)
person.head()

person_id  age citizen_status nativity_status  weekly_hours_worked  \
0         10   62  citizen_birth     native_born                   55   
1         11   65  citizen_birth     native_born                   40   
2         15   70  citizen_birth     native_born                    0   
3         16   70  citizen_birth     native_born                   20   
4         43   75  citizen_birth     native_born                    0   

  marital_status school_status     sex     when_last_worked worked_last_week  \
0        married   not_student  FEMALE         in_last_year           worked   
1        married   not_student       M         in_last_year           worked   
2        married   not_student    MALE  over_five_years_ago     did_not_work   
3        married   not_student       F         in_last_year     did_not_work   
4        married   not_student  FEMALE  over_five_years_ago     not_reported   

  language  race_native_american  race_asian  race_black  \
0  English                     0           0           0   
1  English                     0           0           0   
2  English                     0           0           0   
3  English                     0           0           0   
4  English                     0           0           0   

   race_native_hawaiian  race_pacific_islander  race_white  race_other  \
0                     0                      0           1           0   
1                     0                      0           1           0   
2                     0                      0           1           0   
3                     0                      0           1           0   
4                     0                      0           1           0   

   uninsured  household_id  hh_income_pct  
0          0             5       0.389078  
1          0             5       0.610922  
2          0             9       0.735381  
3          0             9       0.264619  
4          0            22       0.250230

hh_income_pct:
  description:
    - Percentage of the household's total income that comes from that person; 
      calculated as (person's total income) / (household income).

the total income earned per person; this can
be calculated, however, by multiplying the person’s percent of household
income (hh_income_pct) with the household income (hh_income) found in the
household-level data set.

In [11]:
new = pd.merge(person, household, how='inner')

new.head()

MergeError: No common columns to perform merge on. Merge options: left_on=None, right_on=None, left_index=False, right_index=False

In [10]:
new['personal_income'] = new.hh_income_pct*new.hh_income

new = new.sort_values(by='personal_income', ascending=False)
new.head(3)

NameError: name 'new' is not defined

Personal id = 23355, Income = $1060273


Personal id = 89429, Income = $840245


Personal id = 82159, Income = $815297

## Calculate the 80th percentile of age among people in each of the five marital categories.


divorced         67.0


married          67.0


never_married    45.0


separated        61.0


widowed          86.0

In [ ]:
new.marital_status.value_counts()


In [7]:
new.groupby('marital_status').age.quantile(0.8)


NameError: name 'new' is not defined